diff --git a/util/CMakeLists.txt b/util/CMakeLists.txt
index df0efa7..930236b 100644
--- a/util/CMakeLists.txt
+++ b/util/CMakeLists.txt
@@ -154,7 +154,7 @@ if (BUILD_SHARED)

   target_link_libraries (gvm_util_shared LINK_PRIVATE ${GLIB_LDFLAGS}
                          ${GIO_LDFLAGS} ${GPGME_LDFLAGS} ${ZLIB_LDFLAGS}
-                         ${RADIUS_LDFLAGS} ${LIBSSH_LDFLAGS}
+                         ${GNUTLS_LDFLAGS} ${RADIUS_LDFLAGS} ${LIBSSH_LDFLAGS}
                          ${GCRYPT_LDFLAGS} ${LDAP_LDFLAGS} ${REDIS_LDFLAGS}
                          ${UUID_LDFLAGS} ${LINKER_HARDENING_FLAGS})
 endif (BUILD_SHARED)
