diff --git a/src/manage_pg.c b/src/manage_pg.c
index 67f8659..4b25f99 100644
--- a/src/manage_pg.c
+++ b/src/manage_pg.c
@@ -3267,12 +3267,13 @@ check_db_sequences ()
 {
   iterator_t sequence_tables;
   init_iterator (&sequence_tables,
+                 "WITH serial_sequences AS ("
                  "SELECT table_name, column_name,"
-                 "       pg_get_serial_sequence (table_name, column_name)"
+                 "       pg_get_serial_sequence (table_name, column_name) p"
                  "  FROM information_schema.columns"
-                 "  WHERE table_schema = 'public'"
-                 "    AND pg_get_serial_sequence (table_name, column_name)"
-                 "        IS NOT NULL;");
+                 "  WHERE table_schema = 'public')"
+                 " SELECT * FROM serial_sequences"
+                 "  WHERE p IS NOT NULL;");

   while (next (&sequence_tables))
     {
